pycode/memilio-epidata/memilio/epidata_test/test_epidata_modifyDataframeSeries.py

```diff
@@ -647,37 +647,78 @@ def test_interval_mapping(self):
                 self.assertAlmostEqual(test_val[0], calculated_val[0])
 
     def test_fit_age_group_intervals(self):
-        # testing refinement
-        from_age_1 = ["1-10 years", "11-60 years", "61-99 years"]
-        population_1 = [4, 10, 8]
+        age_in_1 = []
+        age_in_1.append(["1-10 years", "11-60 years", "61-99 years"])
+        age_in_1.append([4, 10, 8])
+        df_age_in_1 = pd.DataFrame(age_in_1)
         to_age_1 = ["1-5", "6-10", "11-50", "51-99"]
 
-        from_age_2 = ["1-10", "11-60", "61-80", ">80 years"]
-        population_2 = [4, 5, 8, 5]
+        age_in_2 = []
+        age_in_2.append(["1-10", "11-60", "61-80", ">80 years"])
+        age_in_2.append([4, 5, 8, 5])
+        df_age_in_2 = pd.DataFrame(age_in_2)
         to_age_2 = ["1-99"]
 
-        from_age_3 = ["1-99 years"]
-        population_3 = [0]
-        to_age_3 = ["1-99"]
+        age_in_3 = []
+        age_in_3.append(["0-9", "10-85", ">85"])
+        age_in_3.append([0, 0, 0])
+        df_age_in_3 = pd.DataFrame(age_in_3)
+        to_age_3 = ["0-99"]
 
-        from_age_4 = ["0-9", "10-99", ">100"]
-        population_4 = [4, 10, 8]
-        to_age_4 = ["0-9", "10-19", "20-89"]
+        age_in_4 = []
+        age_in_4.append(["1-99 years"])
```

Comment:
We should pay attention with the max/min. see above.

Reply:
also added with the latest commit. Is now more dynamic.